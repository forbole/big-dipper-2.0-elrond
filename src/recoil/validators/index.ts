import {
  atomFamilyState,
} from './atom';

import {
  readValidator,
  writeValidator,
} from './selectors';

export {
  atomFamilyState,
  readValidator,
  writeValidator,
};
