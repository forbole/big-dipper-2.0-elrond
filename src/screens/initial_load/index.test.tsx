import React from 'react';
import renderer from 'react-test-renderer';
import { RecoilRoot } from 'recoil';
import {
  MockTheme, wait,
} from '@tests/utils';
import InitialLoad from '.';

// ==================================
// unit tests
// ==================================
describe('screen: InitialLoad', () => {
  it('matches snapshot', async () => {
    let component;
    renderer.act(() => {
      component = renderer.create(
        <RecoilRoot>
          <MockTheme>
            <InitialLoad />
          </MockTheme>
        </RecoilRoot>,
      );
    });
    await wait();

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  afterEach(() => {
    jest.clearAllMocks();
  });
});
