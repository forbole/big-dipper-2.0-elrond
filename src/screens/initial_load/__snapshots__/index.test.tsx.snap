// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`screen: InitialLoad matches snapshot 1`] = `
<div
  className="makeStyles-root"
>
  <div>
    <img
      alt="logo"
      className="makeStyles-logo"
      src="https://raw.githubusercontent.com/forbole/big-dipper-assets/master/elrond/logo.svg?sanitize=true"
    />
    <div
      className="MuiLinearProgress-root MuiLinearProgress-colorPrimary makeStyles-divider MuiLinearProgress-indeterminate"
      role="progressbar"
    >
      <div
        className="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Indeterminate"
        style={Object {}}
      />
      <div
        className="MuiLinearProgress-bar MuiLinearProgress-bar2Indeterminate MuiLinearProgress-barColorPrimary"
        style={Object {}}
      />
    </div>
    < />
  </div>
</div>
`;
